### PR TITLE
(FB: 155493) - Allow suppression of Invoices

### PR DESCRIPTION
**Pull request by benrudolph**
This hides invoices from reports and has a management command to hide
or unhide the invoices.

http://manage.dimagi.com/default.asp?155493#869448

### CODE REVIEW

File: corehq/apps/accounting/tests/test_invoicing.py [hunk 1]
```diff
@@ -398,3 +399,23 @@ def _delete_sms_billables(self):
...
         SmsUsageFee.objects.all().delete()
         SmsUsageFeeCriteria.objects.all().delete()
+
+
+class TestManagementCmdInvoice(BaseInvoiceTestCase):
+    def test_hide_invoices(self):
+        """
+        Tests hiding invoices via the management command
+        """
+        invoice_date = utils.months_from_date(self.subscription.date_start, random.randint(2, self.subscription_length))
```
Comment (dimagimon) [inline corehq/apps/accounting/tests/test_invoicing.py]: E501 line too long (120 > 115 characters)

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
assume this is for @nickpell or @biyeun. funny/odd there's no UI. wonder if we'll make one ever

**Comment by benrudolph** [comment]
mainly no UI for it b/c the ticket seemed to be a one-off procedure. instead of deleting the reports from the database, it may be wiser to just hide them (in case something goes wrong). we can discuss

File: corehq/apps/accounting/management/commands/hide_invoices_by_id.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Use modern Python
+from __future__ import unicode_literals, absolute_import, print_function
```
Comment (nickpell) [inline corehq/apps/accounting/management/commands/hide_invoices_by_id.py]: somebody wants to convert to 3.0 ;)

File: corehq/apps/accounting/migrations/0033_auto__add_field_invoice_is_hidden_to_ops.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
+# encoding: utf-8
```
Comment (nickpell) [inline corehq/apps/accounting/migrations/0033_auto__add_field_invoice_is_hidden_to_ops.py]: sorry, I already added migration 0033. You can remove this migration, rebase on master, and then re-create the migrations to get 0035.

File: corehq/apps/accounting/management/commands/hide_invoices_by_id.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+class Command(BaseCommand):
+    help = 'Hides the specified invoice(s) from showing on reports'
+
+    option_list = BaseCommand.option_list + (
+        make_option('-u', '--unhide',
+                    action='store_true',
+                    default=False,
+                    dest='unhide',
+                    help="Make invoice(s) visible to the operations team"
```
Comment (nickpell) [inline corehq/apps/accounting/management/commands/hide_invoices_by_id.py]: make visible or invisible?

File: corehq/apps/accounting/tests/test_invoicing.py [hunk 5]
```diff
@@ -398,3 +399,24 @@ def _delete_sms_billables(self):
         SmsGatewayFeeCriteria.objects.all().delete()
         SmsUsageFee.objects.all().delete()
         SmsUsageFeeCriteria.objects.all().delete()
+
+
+class TestManagementCmdInvoice(BaseInvoiceTestCase):
```
Comment (nickpell) [inline corehq/apps/accounting/tests/test_invoicing.py]: :+1: tests

**Comment by nickpell** [comment]
a couple of comments to address, then good to merge

File: corehq/apps/accounting/management/commands/hide_invoices_by_id.py (hunk 4 above)
Comment (benrudolph) [inline corehq/apps/accounting/management/commands/hide_invoices_by_id.py]: This is actually supposed to be visible. The option basically reverses what the command normally does.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/accounting/management/commands/hide_invoices_by_id.py (hunk 4 above)
Comment (nickpell) [inline corehq/apps/accounting/management/commands/hide_invoices_by_id.py]: oh right

**Comment by nickpell** [comment]
will merge when tests pass

**Pull request merged**